### PR TITLE
fcrypt: save timestamps of config files

### PR DESCRIPTION
**Pull request by petermax2**
# Purpose

The resolver thinks that the files were modifed externally when fcrypt overwrites them.
So we restore the original timestamps after fcrypt modified the file.

See #1050 for further details.

# Checklist

Please only check relevant points.
For docu fixes, spell checking and similar nothing
needs to be checked.

- [x] commit messages are fine (with references to issues)
- [x] I ran all tests and everything went fine
- [ ] I added unit tests
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions
- [ ] meta data is updated (e.g. README.md of plugins)

@markus2330 please review my pull request

### CODE REVIEW

File: src/plugins/fcrypt/fcrypt.c [hunk 1]
```diff
@@ -131,6 +131,48 @@ static size_t getRecipientCount (KeySet * config)
...
+/**
+ * @brief Determines mtime for the file given by parentKey.
+ * @param parentKey holds the path to the file as string value. Is also used for storing warnings.
+ * @param fileStat will hold the mtime on success.
+ * @retval 1 on success
+ * @retval 0 on failure. In this case a warning is appended to parentKey.
+ */
+static int fcryptSaveMtime (Key * parentKey, struct stat * fileStat)
+{
+	if (stat (keyString (parentKey), fileStat) == -1)
```
Comment (petermax2) [inline src/plugins/fcrypt/fcrypt.c]: Check if `keyString(parentKey)` is a valid file, otherwise the warning is set.

**Comment by markus2330** [comment]
Is this ready to merge besides that the build server takes forever today (at least the builds started now)?

**Comment by petermax2** [comment]
Yes it is theoretically ready. Did it solve the make run_all issue?

**Comment by markus2330** [comment]
Thanks! I will open a new issue for the run_all if it still exists.

**Pull request merged**